streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -229,22 +240,24 @@ private void updateProcessorContext(final StampedRecord record, final ProcessorN
 
     /**
      * <pre>
-     *  - flush state and producer
-     *  - write checkpoint
-     *  - commit offsets
+     * - flush state and producer
+     * - if(!eos) write checkpoint
+     * - commit offsets and maybe start new transaction
      * </pre>
      */
     @Override
-    public void commit() {
+    public void commit(final boolean startNewTransaction) {
```

Comment:
This would not work. Even with EOS enabled, we call this method with `true` and `false`: When we commit on `suspend` or `close` we do not want to start a new transaction, while on regular commit (ie, commit interval) we do want to start a new transaction.